plugins/auth-backend-module-microsoft-provider/src/authenticator.ts

```diff
@@ -31,6 +32,10 @@ export const microsoftAuthenticator = createOAuthAuthenticator({
     const clientSecret = config.getString('clientSecret');
     const tenantId = config.getString('tenantId');
     const domainHint = config.getOptionalString('domainHint');
+    const scope = scopeHelper(
+      ['user.read'],
+      config.getOptionalStringArray('additionalScopes'),
+    );
```

Comment:
You could replace `scopeHelper` with `union` from lodash (https://lodash.com/docs/4.17.15#union) or move the 2 lines from `scopes.ts` here. This way you yould remove `scopes.ts` and `scopes.test.ts`. I would like to vouch for not creating a new implementation of concating & deduplicating an array if it is only used once here.

I believe what Patriks comment was aiming for is if `config.getOptionalStringArray('additionalScopes')` actually picks up the scopes from the `app.config.yaml`. Did you try it locally? :) As far as I understand it the test in `module.test.ts` should cover this though.

Reply:
Changed the implementation to use union from loadash.